File: incident-service/src/main/java/edu/asp/uptimearmor/incidentservice/entity/Incident.java
package edu.asp.uptimearmor.incidentservice.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name="incident", schema="uptime_armor")
public class Incident {

    @Id
    @Column(name="incident_id", nullable = false)
    private String incidentId;
    @Column(name="description")
    private String description;
    @Column(name="category")
    private String category;
    @Column(name="creation_date")
    @JsonFormat(pattern ="yyyy-MM-dd'T'HH:mm")
    private Date creationDate;
    @JsonFormat(pattern ="yyyy-MM-dd'T'HH:mm")
    @Column(name="closed_date")
    private Date closedDate;
    @Column(name="severity")
    private int severity;
    @Column(name="priority")
    private int priority;
    @ManyToOne
    @JoinColumn(name="business_unit_id", referencedColumnName = "business_unit_id")
    private BusinessUnit businessUnit;
    @ManyToOne
    @JoinColumn(name="reporter", referencedColumnName="employee_id")
    private Employee reporter;
    @ManyToOne
    @JoinColumn(name="assignee", referencedColumnName="employee_id")
    private Employee assignee;
    @Column(name="status")
    private String status;
    @Column(name="resolution")
    private String resolution;
}

File: uptime-armor-fe/src/app/inc/models/incident.ts
  export class IncidentModel {
    description: string;
    category: string;
    creationDate: Date;
    closedDate: Date;
    severity: number;
    priority: number;
    businessUnit: BusinessUnitIdModel = {
      businessUnitId: '',
      businessUnitName: ''
    };
    reporter: EmployeeIdModel = {
      employeeId: '',
      firstName: '',
      lastName: ''
    };
    assignee: EmployeeIdModel =  {
      employeeId: '',
      firstName: '',
      lastName: ''
    };
    status: string;
    resolution: string;
  }

  export class IncidentEntity extends IncidentModel{
    incidentId: string;
  }
  
  export class BusinessUnitIdModel {
    businessUnitId: string;
    businessUnitName: string;
  }
  
  export class EmployeeIdModel {
    employeeId: string;
    firstName: string;
    lastName: string;

  }

  export class OutageEntity extends IncidentEntity {

  }

  export class IncidentExportToExcel {
    
    constructor(
      public incidentId: string,
      public description: string,
      public category: string,
      public creationDate: Date,
      public closedDate: Date | null,
      public severity: number,
      public priority: number,
      public businessUnit: string,
      public reporter: string,
      public assignee: string,
      public status: string,
      public resolution: string
    ) {}
     
  }

  export class OutageEportToExcel extends IncidentExportToExcel{

  }

